os/services/ip64/ip64-dns64.c

```diff
@@ -244,18 +306,43 @@ ip64_dns64_4to6(const uint8_t *ipv4data, int ipv4datalen,
         ipv6datalen += 12;
 
       } else {
+        if(adata + len > ipv4data + ipv4datalen ) {
```

Comment:
Can't this check be done before the if-statement to reduce the code duplication?

All conditions in the if-statements seem to have a trailing whitespace, which none of the existing code seems to have.

Reply:
Thank you for the comments. Yes, the check can be done before. I had it in this de-duplicated version first, but then duplicated the code to move the check closer to the use of the pointer. However, I will change this back to de-duplicating the check.

I will also remove the trailing whitespaces

Reply:
I applied the changes